src/Association/Embedded.php

```diff
@@ -236,12 +238,13 @@ public function setIndexClass($name)
      * Get/set the index class used for this embed.
      *
      * @deprecated 3.2.0 Use setIndexClass()/getIndexClass() instead.
-     * @param string|null|\Cake\ElasticSearch\Index $name The class name to set.
+     * @param \Cake\ElasticSearch\Index|string|null $name The class name to set.
      * @return string The class name.
      */
-    public function indexClass($name = null)
+    public function indexClass(string|Index|null $name = null): string
```

Comment:
```suggestion
    public function indexClass(Index|string|null $name = null): string
```

The sniff that converts annotations to type declarations runs before the sniff that fixes the annotation order.